pallas-network/src/multiplexer.rs

```diff
@@ -92,25 +98,45 @@ impl Bearer {
         Ok((Self::Unix(stream), addr))
     }
 
+    #[cfg(windows)]
+    pub async fn connect_named_pipe(pipe_name: impl AsRef<std::ffi::OsStr>) -> Result<Self, tokio::io::Error>{
+        let client = loop {
+            match tokio::net::windows::named_pipe::ClientOptions::new().open(&pipe_name) {
+                Ok(client) => break client,
+                Err(e) if e.raw_os_error() == Some(windows_sys::Win32::Foundation::ERROR_PIPE_BUSY as i32) => (),
+                Err(e) => return Err(e),
+            }
+        
+            tokio::time::sleep(std::time::Duration::from_millis(50)).await;
+        };
+        Ok(Self::NamedPipe(client))
+    }
+
     #[cfg(not(target_os = "windows"))]
-    pub async fn connect_unix(path: impl AsRef<Path>) -> Result<Self, tokio::io::Error> {
+    pub async fn connect_unix(path: impl AsRef<std::path::Path>) -> Result<Self, tokio::io::Error> {
         let stream = UnixStream::connect(path).await?;
         Ok(Self::Unix(stream))
     }
 
-    pub async fn readable(&self) -> tokio::io::Result<()> {
+    pub async fn readable(&mut self) -> tokio::io::Result<()> {
         match self {
             Bearer::Tcp(x) => x.readable().await,
             #[cfg(not(target_os = "windows"))]
             Bearer::Unix(x) => x.readable().await,
+
+            #[cfg(target_os = "windows")]
+            Bearer::NamedPipe(x) => x.readable().await
+            
         }
     }
 
-    fn try_read(&mut self, buf: &mut [u8]) -> tokio::io::Result<usize> {
+    async fn try_read(&mut self, buf: &mut [u8]) -> tokio::io::Result<usize> {
```

Comment:
why is the fn signature changing to `async`? is this a requirement for named pipes?

Reply:
nope, not needed - reverting it